app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome, #{@user.username}."
+      redirect_to user_path(@user)
+    else
+      render :new
```

Comment:
@kwokhuen 
Make sure to set the status code here. You can do it as an option to the `render` method

```
render :new, status: 422
```